hw/top_earlgrey/data/top_earlgrey.sv.tpl

```diff
@@ -733,32 +733,64 @@ slice = str(alert_idx+w-1) + ":" + str(alert_idx)
 % if num_dio != 0:
   // Dedicated IO connections
   // Input-only DIOs have no d2p signals
-  assign dio_d2p = {
+  assign dio_d2p = {<% vector_idx = num_dio - 1 %>
   % for sig in top["pinmux"]["dio"]:
     % if sig["type"] in ["output", "inout"]:
-    cio_${sig["name"]}_d2p${"" if loop.last else ","} // DIO${num_dio - 1 - loop.index}
+      % if sig["width"] > 1:
+        % for i in range(sig["width"]-1,-1,-1):
```

Comment:
i'm curious, why not just `sig[width]-1`? is it because you're trying to place the highest indices first?

Reply:
definitely something for another PR..but i wonder if we should wrap these up in separate python functions eventually.. just because so much of the code is repeated, with only the text being different.

It might make it even HARDER to read though..so i'm not super convinced.

Reply:
yeah this needs to be in descending order.

we should definitely start to factor out some of these things into functions.
this is mostly a carry-over from Bronze to get things going.